app/javascript/src/utils/project.js

```diff
@@ -44,12 +44,22 @@ export function createDemoProject(title) {
 export async function fetchProject(uuid) {
   const baseUrl = "/projects/"
 
+  const localProject = getProjectFromLocalStorage(uuid)
+
   return await new FetchRails(baseUrl + uuid).get()
     .then(data => {
       if (!data) throw Error("No results")
 
       const parsedData = JSON.parse(data)
 
+      // If the project in localStorage is newer than the project from the API
+      // something may have gone wrong when saving to the API, but the localStorage
+      // was still saved as expected. In this case we use the data from localStorage.
+      // This is a fallback and should not be the norm.
+      if (localProject && new Date(parsedData.updated_at) < new Date(localProject.updated_at)) {
+        parsedData.content = localProject.content
```

Comment:
I wonder if we could send an event over to Bugsnag here or something so we can track how often this incident happens?

Reply:
Good idea! We don't have bugsnag set up for javascript currently, but I will have a look

Reply:
Turns out it was fairly easy, their javascript package is pretty nice! I will add this to other parts of the editor as well, but I will save that for another PR